docs: add release notes for 0.27.0 (#8671)
determined-ai/determined: docs/release-notes.rst

 Release Notes
###############

**************
 Version 0.27
**************

Version 0.27.0
==============

**Release Date:** January 09, 2024

**Breaking Changes**

> -  Experiments: Allow empty model definitions when creating experiments.

Reviewer: This seems like maybe it shouldn't count as breaking. @corban-beaird?

Reviewer 2: Yea, that's a fair point, the second bullet point is the breaking change, the first is the reason why the break is occuring

Author: @corban-beaird should the first bullet be removed all together or moved to the `improvements` section?

> -  CLI: Optional flags must come before or after positional arguments when creating experiments,
>    orderings such as the following are no longer supported ``det e create const.yaml -f .``, instead
>    use ``det e create -f const.yaml .`` or ``det e create const.yaml . -f``.

Reviewer: ```suggestion
-  CLI: Optional flags must come before or after positional arguments when creating experiments;
   orderings such as ``det e create const.yaml -f .`` are no longer supported. Instead, you should
   use ``det e create -f const.yaml .`` or ``det e create const.yaml . -f``.
```

**Improvements**

> -  Allow checkpoint downloads through the server for ``checkpoint_storage`` type ``shared_fs`` and

Reviewer: ```suggestion
-  Allow checkpoint downloads through the server for ``checkpoint_storage`` types ``shared_fs`` and
```
